pkg/storage/gcp/gcp.go

```diff
@@ -0,0 +1,78 @@
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"cloud.google.com/go/datastore"
+	"cloud.google.com/go/storage"
+	"github.com/gobuffalo/envy"
+	multierror "github.com/hashicorp/go-multierror"
+	"google.golang.org/api/option"
+)
+
+// Storage implements the (github.com/gomods/pkg/storage).Saver interface
+type Storage struct {
+	bucket       *storage.BucketHandle
+	dsClient     *datastore.Client
+	baseURI      *url.URL
+	closeStorage func() error
+}
+
+// NewWithCredentials returns a new Storage instance authenticated using the provided
+// ClientOptions. The bucket name to be used will be loaded from the
+// environment variable ATHENS_STORAGE_GCP_BUCKET.
+// TODO: project ID for datastore
+//
+// The ClientOptions should provide permissions sufficient to read, write and
+// delete objects in google cloud storage for your project.
+// TODO: As well as datastore
+func NewWithCredentials(ctx context.Context, cred option.ClientOption) (*Storage, error) {
+	storage, err := storage.NewClient(ctx, cred)
+	if err != nil {
+		return nil, fmt.Errorf("could not create new storage client: %s", err)
+	}
+	bucketname, err := envy.MustGet("ATHENS_STORAGE_GCP_BUCKET")
```

Comment:
env.GcpBucketName

Reply:
@michalpristas I don't see that function anywhere. Did I miss it?